### PR TITLE
Change util.EncryptConfig,DecryptConfig, GetFormat and make it public

### PR DESCRIPTION
**Pull request by rhatdan**
We want to share these functions with Podman, Podman currently has a slightly different version which is correct, so use correct version in Buildah and vendor it into Podman.

Fixing: https://github.com/containers/podman/issues/18196

#### What type of PR is this?

> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test 
> /kind feature
> /kind flake
> /kind other

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by rhatdan** [comment]
This is an alternative to https://github.com/containers/buildah/pull/4745

**Comment by rhatdan** [comment]
Once this merges we can vendor it into Podman and then share the code.

**Comment by mtrmac** [comment]
Sharing that would be nice, but let’s make sure we have a consensus on what the right thing to do is.

Most of the conversation is in https://github.com/containers/podman/issues/18196 .

**Comment by mtrmac** [comment]
(A “make … public” PR that actually, _primarily_, makes a behavior change, would be hard to track down later, so renaming it.)

File: pkg/cli/build.go [hunk 1]
```diff
@@ -272,8 +271,8 @@ func GenBuildOptions(c *cobra.Command, inputArgs []string, iopts BuildOptions) (
 		return options, nil, nil, err
 	}
 
-	decryptConfig, err := iutil.DecryptConfig(iopts.DecryptionKeys)
-	if err != nil {
+	decryptConfig, err := util.DecryptConfig(iopts.DecryptionKeys)
+	if err != nil || decryptConfig == nil {
```
Comment (mtrmac) [inline pkg/cli/build.go:275]: `decryptConfig == nil` means “don’t decrypt”, and it’s what I propose we do by default; failing on that doesn’t make sense to me.

File: tests/pull.bats [hunk 2]
```diff
@@ -190,7 +190,7 @@ load helpers
   run_buildah push $WITH_POLICY_JSON --encryption-key jwe:${TEST_SCRATCH_DIR}/tmp/mykey.pub busybox  oci:${TEST_SCRATCH_DIR}/tmp/busybox_enc
 
   # Try to pull encrypted image without key should fail
-  run_buildah 125 pull $WITH_POLICY_JSON oci:${TEST_SCRATCH_DIR}/tmp/busybox_enc
+  run_buildah 1 pull $WITH_POLICY_JSON oci:${TEST_SCRATCH_DIR}/tmp/busybox_enc
```
Comment (TomSweeneyRedHat) [inline tests/pull.bats:193]: Can we add tests to these when a key isn't available so it can still test a failure path?

Comment (rhatdan) [reply tests/pull.bats:193]: This is testing a failure path.

File: pkg/util/util.go [hunk 3]
```diff
@@ -77,3 +81,49 @@ func DiscoverContainerfile(path string) (foundCtrFile string, err error) {
...
+			return nil, nil, fmt.Errorf("invalid encryption keys: %w", err)
+		}
+		cc := encconfig.CombineCryptoConfigs([]encconfig.CryptoConfig{ecc})
+		encConfig = cc.EncryptConfig
+	}
+	return encConfig, encLayers, nil
+}
+
+// GetFormat translates format string into either docker or OCI format constant
+func GetFormat(format string) (string, error) {
```
Comment (mtrmac) [inline pkg/util/util.go:120]: Should this be a part of the move at all?

**Review by mtrmac — COMMENTED** [review]
Please carry over the test suite updates from https://github.com/containers/buildah/pull/4745 to make this pass.

**Comment by mtrmac** [comment]
@rhatdan this is failing tests.

**Comment by rhatdan** [comment]
@mtrmac PTAL I think I followed all of your instructions.

**Review by mtrmac — COMMENTED** [review]
/lgtm

**Comment by edsantiago** [comment]
This is [breaking podman](https://api.cirrus-ci.com/v1/artifact/task/6125041562157056/html/bud-podman-fedora-38-root-host-boltdb.log.html):
```
[+1068s] not ok 269 bud with encrypted FROM image
...
# #| FAIL: exit code is 125; expected 1
```
And also, the podman message is still "missing private key", whereas the changed test now expects "invalid tar header"

**Pull request merged**